### PR TITLE
Create configuration page for lights

### PR DESCRIPTION
**Pull request by Williangalvani**
image

### CODE REVIEW

File: core/frontend/src/components/vehiclesetup/configuration/lights.vue [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+          if (old_param.name === new_param_name) {
+            continue
+          }
+          if (old_param.value === Lights.Lights2) {
+            // set the old parameter to DISABLED
+            mavlink2rest.setParam(old_param.name, Lights.DISABLED, autopilot_data.system_id)
+          }
+        }
+        mavlink2rest.setParam(new_param_name, Lights.Lights2, autopilot_data.system_id)
+      }
```
Comment (patrickelectric) [inline core/frontend/src/components/vehiclesetup/configuration/lights.vue:134]: Can we encapsulate this in a single function for both lights ?

File: core/frontend/src/components/vehiclesetup/configuration/lights.vue [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+          :item-text="friendlyName"
+          :item-value="'name'"
+          label="Lights 2"
+        />
+      </v-card-text>
+    </v-card>
+    <v-card>
+      <v-card-title> Joystick steps </v-card-title>
+      <v-card-text>
+        How many button presses it takes to go from 0% to 100% brightness.
```
Comment (patrickelectric) [inline core/frontend/src/components/vehiclesetup/configuration/lights.vue:32]: I believe that asking the user a percentage per step over the number of buttons press is more friendly. Maybe rephrasing it helps.

**Review by patrickelectric — CHANGES_REQUESTED** [review]

**Comment by Williangalvani** [comment]
updated

File: core/frontend/src/components/vehiclesetup/configuration/lights.vue (hunk 2 above)
Comment (Williangalvani) [reply core/frontend/src/components/vehiclesetup/configuration/lights.vue:32]: improved the wording and added a live-updated percentage

File: core/frontend/src/components/vehiclesetup/configuration/lights.vue (hunk 1 above)
Comment (Williangalvani) [reply core/frontend/src/components/vehiclesetup/configuration/lights.vue]: done

File: core/frontend/src/components/vehiclesetup/configuration/lights.vue [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  DISABLED = 0,
+}
+
+export default {
+  name: 'LightsConfigration',
+  data() {
+    return {
+      lights1_new_param: undefined as (string | undefined), // item-value must be a primitive
+      lights2_new_param: undefined as (string | undefined),
+      steps_new_value: 0,
```
Comment (patrickelectric) [inline core/frontend/src/components/vehiclesetup/configuration/lights.vue:65]: could we use a default value, like 10 ?
Other than that is just weird to start with 0 steps.

**Review by patrickelectric — CHANGES_REQUESTED** [review]

File: core/frontend/src/components/vehiclesetup/configuration/lights.vue (hunk 3 above)
Comment (Williangalvani) [reply core/frontend/src/components/vehiclesetup/configuration/lights.vue]: that should never happen, as we set it to the parameter value both when mounting (if the param is available) or when we detect a parameter change in the watcher. but I did update it anyway.

Comment (patrickelectric) [reply core/frontend/src/components/vehiclesetup/configuration/lights.vue]: When you configure it for the first time, if the lights are not configured, it starts with 0.

**Comment by ES-Alexander** [comment]
Can we add a slider for the steps?
And min/max PWM limits?

Could also be nice to have a little graph showing the steps visually (with display options for both linear and Lumen-exponential brightness outputs).

It might make sense to have the channel controls in the top row, and then the steps slider + graph in a second row (instead of the current approach with everything in a single row).

Thinking something like this:
image

**Comment by patrickelectric** [comment]
There is also another thing, the page end up showing for rover as well, should we remove it ?

**Comment by Williangalvani** [comment]
> Can we add a slider for the steps? And min/max PWM limits?
> 
> Could also be nice to have a little graph showing the steps visually (with display options for both linear and Lumen-exponential brightness outputs).
> 
> It might make sense to have the channel controls in the top row, and then the steps slider + graph in a second row (instead of the current approach with everything in a single row).
> 
> Thinking something like this: image

Fancy!
Idk.
I'm 50% "we can do this later as there's more calibration/setup stuff to be done" and 50% "let's do it now or it won't get done".
@patrickelectric thoughts?

**Review by patrickelectric — APPROVED** [review]
We can move further required changes to issues.

**Pull request merged**